boxsdk/object/webhook.py

```diff
@@ -0,0 +1,47 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+import json
+from .base_object import BaseObject
+from boxsdk.config import API
+from boxsdk.util.translator import Translator
+from ..pagination.marker_based_object_collection import MarkerBasedObjectCollection
+
+
+def _compute_signature(body, headers, signature_key):
+    if signature_key is None:
+        return None
+# convert to braces
+    if headers.box-signature-version != '1':
+        return None
+
+    if headers.box-signature-algorithm != '1':
+        return None
+
+    new_hmac = hmac.new('sha256', signature_key)
+    new_hmac.update(body)
+    new_hmac.update(headers.box-delivery-timestamp)
```

Comment:
I forgot to mention that validate_message() and _compute_signature() methods are still work in progress. Was going to collect feedback first for all the other changes first because there is going to multiple pass throughs of this, sorry!